packages/core/src/conflict-resolution.ts

```diff
@@ -190,32 +194,24 @@ export async function fetchLog(
   const commit = await dispatcher.retrieveCommit(cid)
   if (!commit) throw new Error(`No commit found for CID ${cid.toString()}`)
 
-  let payload = commit
-  const isSignedCommit = StreamUtils.isSignedCommit(commit)
-  if (isSignedCommit) {
-    payload = await dispatcher.retrieveCommit(commit.link)
-    if (!payload) throw new Error(`No commit found for CID ${commit.link.toString()}`)
-  }
-  const prevCid: CID = payload.prev
-  if (!prevCid) {
-    // this is a fake log
-    return []
-  }
+  let prevCid: CID = commit.prev
   let nextEntry: LogEntry
-  if (StreamUtils.isAnchorCommit(commit)) {
+  if (StreamUtils.isSignedCommit(commit)) {
+    const linkedCommit = await dispatcher.retrieveCommit(commit.link)
+    if (!linkedCommit) throw new Error(`No commit found for CID ${commit.link.toString()}`)
+    nextEntry = { cid: cid, type: CommitType.SIGNED, commit: linkedCommit, envelope: commit, timestamp: timestamp }
+    prevCid = linkedCommit.prev
+  } else if (StreamUtils.isAnchorCommit(commit)) {
     const proof = await dispatcher.retrieveFromIPFS(commit.proof)
     timestamp = proof.blockTimestamp
-    nextEntry = {
-      type: CommitType.ANCHOR,
-      cid: cid,
-      timestamp: timestamp,
-    }
+    nextEntry = { cid: cid, type: CommitType.ANCHOR, commit: commit, timestamp: timestamp }
   } else {
-    nextEntry = {
-      type: CommitType.SIGNED,
-      cid: cid,
-      timestamp: timestamp,
-    }
+    // For all cases not using DagJWS for signing (e.g. CAIP-10 links)
+    nextEntry = { cid: cid, type: CommitType.SIGNED, commit: commit, timestamp: timestamp }
+  }
+  if (!prevCid) {
+    // this is a fake log
```

Comment:
any idea what this means?  Is this just supporting unit tests?

Reply:
LOL I was wondering the same thing. Can this condition occur when the genesis commit is processed?

Reply:
This means that someone sent a tip that is a fake log. I.e. a log that at some point does not point to a previous commit or genesis.

Reply:
Ah yes, that makes sense, thanks!  Maybe you could expand the comment a little while we're in there Mohsin?